### PR TITLE
Calculate column stats when broken column threshold is specified.

### PR DESCRIPTION
**Pull request by michelle192837**
This helps flag broken columns (in tab state, where we have broken column thresholds specified), and will later help calculate column alerts, and exclude broken columns from row alerts.

### CODE REVIEW

File: pkg/updater/updater.go [hunk 1]
```diff
@@ -1082,6 +1082,9 @@ func ConstructGrid(log logrus.FieldLogger, cols []InflatedColumn, issues map[str
 	}
 
 	for _, col := range cols {
+		if col.Column != nil {
```
Comment (fejta) [inline pkg/updater/updater.go:1085]: Is it worth putting this new behavior behind a flag?

**Review by fejta — APPROVED** [review]
/hold

File: pkg/updater/updater.go (hunk 1 above)
Comment (michelle192837) [reply pkg/updater/updater.go:1085]: I was debating on it, but yeah might as well!

Comment (fejta) [reply pkg/updater/updater.go:1085]: Is this a new field? If so then maybe not so important. Does it even appear in the UI?

Comment (fejta) [reply pkg/updater/updater.go:1085]: But still IMO good practice!~

Comment (michelle192837) [reply pkg/updater/updater.go:1085]: New field, doesn't appear in the UI yet (though maybe it'll be a nice FixIt item to flag columns as broken, not quite trivial due to how column headers get combined rn). Given this is a shared updater function, would it make sense to flag this in tabulator:main + use `brokenThreshold > 0.0` explicitly here?

Comment (fejta) [reply pkg/updater/updater.go:1085]: Sure, the goal here is mostly to design the failure, maybe when we encounter "this is for mysterious reasons causing testgrid to crash" or "all the updaters are surprisingly slow after this push" that we have a cheap way to turn this functionality off while we investigate.

File: pkg/updater/updater.go [hunk 2]
```diff
@@ -1419,6 +1422,42 @@ func alertInfo(failures int32, msg, cellID, latestCellID string, userProperties
...
+		default:
+			// blank cell or unrecognized status, do nothing
+		}
+	}
+	return &statepb.Stats{
+		FailCount:  fails,
+		PassCount:  passes,
+		TotalCount: total,
+		Pending:    pending,
+		Broken:     float32(fails)/float32(total) > brokenThreshold,
```
Comment (fejta) [inline pkg/updater/updater.go:1457]: if total is 0 this will blow up, for example

Comment (michelle192837) [reply pkg/updater/updater.go:1457]: Daaaaaangit.

**Review by fejta — APPROVED** [review]

**Pull request merged**